redbeams/src/main/java/com/sequenceiq/redbeams/service/stack/DBStackUpdater.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.redbeams.service.stack;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.redbeams.api.endpoint.v4.databaseserver.responses.SslCertificateType;
+import com.sequenceiq.redbeams.configuration.DatabaseServerSslCertificateConfig;
+import com.sequenceiq.redbeams.configuration.SslCertificateEntry;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.domain.stack.SslConfig;
+
+@Service
+public class DBStackUpdater {
+
+    @Inject
+    private DatabaseServerSslCertificateConfig databaseServerSslCertificateConfig;
+
+    @Inject
+    private DBStackService dbStackService;
+
+    public void updateSslConfig(long id) {
+        Optional<DBStack> dbStackOpt = dbStackService.findById(id);
+        dbStackOpt.ifPresent(dbStack -> {
+            SslConfig sslConfig = dbStack.getSslConfig();
```

Comment:
Cannot remember how legacy db stacks are handled, but I fear `sslConfig` may be `null`. So I would add a separate logic to initialize it if needed (e.g. as `Optional.ofNullable(dbStack.getSslConfig()).orElse(new SslConfig())`).